File: frontend/src/features/prediction/types/prediction.ts
export type AQICategory = 'Good' | 'Moderate' | 'Poor' | 'Very Poor' | 'Severe' | 'Hazardous';
export type HealthCondition = 'none' | 'asthma' | 'copd' | 'heart_disease' | 'elderly' | 'children' | 'pregnant';

export interface PredictionRequest {
  latitude: number;
  longitude: number;
  prediction_time: string;
  city?: string;
  health_condition?: HealthCondition;
}

export interface PredictionResponse {
  predicted_aqi: number;
  aqi_category: AQICategory;
  catboost_prediction: number;
  lightgbm_prediction: number;
  xgboost_prediction: number;
  ensemble_prediction: number;
  confidence_score: number;
  prediction_latency_ms: number;
  model_version: string;
  ensemble_version: string;
  prediction_timestamp: string;
  prediction_source: string;
  health_risk_level: string;
  pm25: number;
  pm10: number;
  so2?: number;
  no2?: number;
  temperature: number;
  humidity: number;
  wind_speed: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from typing import Optional
from app.models.enums import AQICategory, HealthCondition

class PredictionRequest(BaseModel):
    latitude: float = Field(..., ge=-90, le=90)
    longitude: float = Field(..., ge=-180, le=180)
    prediction_time: datetime
    city: Optional[str] = None
    health_condition: Optional[HealthCondition] = None
    skip_history: bool = False
    user_id: Optional[str] = None

class PredictionResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    predicted_aqi: float
    aqi_category: AQICategory
    catboost_prediction: float
    lightgbm_prediction: float
    xgboost_prediction: float
    ensemble_prediction: float
    confidence_score: float
    prediction_latency_ms: float
    model_version: str
    ensemble_version: str
    prediction_timestamp: datetime
    prediction_source: str
    health_risk_level: str
    pm25: float = 0.0
    pm10: float = 0.0
    no2: float = 0.0
    so2: float = 0.0
    o3: float = 0.0
    co: float = 0.0
    temperature: float = 0.0
    humidity: float = 0.0
    wind_speed: float = 0.0
